src/django_idom/hooks.py

```diff
@@ -24,13 +27,30 @@
 
 
 def use_location() -> Location:
-    """Get the current route as a string"""
+    """Get the current route as a `Location` object"""
     # TODO: Use the browser's current page, rather than the WS route
     scope = use_scope()
     search = scope["query_string"].decode()
     return Location(scope["path"], f"?{search}" if search else "")
 
 
+def use_origin() -> str | None:
```

Comment:
I think we should limit what hooks we include to those that solve problems specific to the Django framework. This seems generally useful - something that could belong in a package that anyone using IDOM could take advantage of.

Reply:
The `use_origin` hook is specific to Django Channels. It's something I determined would be frequently used when creating SPAs with `django`+`idom`.

This is fairly easy to obtain when using Django HTML templates (http), but is very awkward to obtain when using IDOM (websockets) for rendering.